### ISSUE TITLE
/opt folder missing

### ISSUE THREAD

**Issue by FrenZy931** [Botspot/pi-apps#2466]
### Confirmations

- [X] I searched for similar bug reports in https://github.com/Botspot/pi-apps/issues and found none were relevant.
- [X] I tried updating Pi-Apps and the issue is still not fixed. [How to update Pi-Apps?](https://pi-apps.io/wiki/getting-started/updating-pi-apps/)
- [X] IMPORTANT: I promise to respond to comments on this issue and write a good bug report https://www.softwaretestinghelp.com/how-to-write-good-bug-report/

### What happened?

App wont installl

### Description

I wanted to install wine64, but i got an error

### What are your system specs (run the following command in your terminal)?

```bash
OS: Ubuntu 23.04
OS architecture: 64-bit
Last updated Pi-Apps on: 10/14/2023
Latest Pi-Apps version: 10/16/2023
Kernel: aarch64 6.2.0-1014-raspi
Device model: Raspberry Pi 4 Model B Rev 1.5 
Cpu name: Cortex-A72
Ram size: 7.38 GB
Language: en_US.UTF-8
```

### (Recommended) Error log? Terminal output? Debug messages?

```bash
Upgrading Box64 if necessary:
sudo: a password is required
Running sudo apt update...
Hit:1 http://ports.ubuntu.com/ubuntu-ports lunar InRelease
Hit:2 http://ports.ubuntu.com/ubuntu-ports lunar-updates InRelease
Hit:3 http://packages.microsoft.com/repos/code stable InRelease
Hit:4 http://ports.ubuntu.com/ubuntu-ports lunar-backports InRelease
Hit:5 http://ports.ubuntu.com/ubuntu-ports lunar-security InRelease
Hit:6 https://Pi-Apps-Coders.github.io/box64-debs/debian ./ InRelease
Hit:7 https://Pi-Apps-Coders.github.io/box86-debs/debian ./ InRelease
All packages are up to date.
apt update complete.

WARNING: apt does not have a stable CLI interface. Use with caution in scripts.

Reading package lists...
Building dependency tree...
Reading state information...
Skipping box64-android, it is not installed and only upgrades are requested.
Skipping box64-generic-arm, it is not installed and only upgrades are requested.
box64-rpi4arm64 is already the newest version (0.2.5+20231017.f67d070-1).
Skipping box64-rpi3arm64, it is not installed and only upgrades are requested.
Skipping box64-tegrax1, it is not installed and only upgrades are requested.
Skipping box64-rk3399, it is not installed and only upgrades are requested.
Skipping box64-generic-arm-page16k, it is not installed and only upgrades are requested.
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
Installed Box64 version:
Box64 with Dynarec v0.2.5 f67d0709 built on Oct 17 2023 06:37:50
Will install these packages: cabextract p7zip-full 
Creating an empty apt-package to install the necessary apt packages...
It will be named: pi-apps-112b87c1
The pi-apps-112b87c1 package is already installed. Inheriting its dependencies: cabextract, p7zip-full
Depends: cabextract, p7zip-full
pi-apps-112b87c1 is already installed and no changes would be made. Skipping...
Package installation complete.
Downloading wine-8.17.tar.gz to /tmp/wine-8.17.tar.gz... 
848KiB/372MiB 
5.4MiB/372MiB 
10MiB/372MiB —
14MiB/372MiB ——
19MiB/372MiB ———
23MiB/372MiB ————
28MiB/372MiB ————
33MiB/372MiB —————
38MiB/372MiB ——————
43MiB/372MiB ———————
48MiB/372MiB ————————
49MiB/372MiB ————————
52MiB/372MiB —————————
56MiB/372MiB ——————————
60MiB/372MiB ——————————
65MiB/372MiB ———————————
70MiB/372MiB ————————————
74MiB/372MiB —————————————
78MiB/372MiB ——————————————
83MiB/372MiB ——————————————
87MiB/372MiB ———————————————
92MiB/372MiB ————————————————
94MiB/372MiB ————————————————
98MiB/372MiB —————————————————
103MiB/372MiB —————————————————
108MiB/372MiB ———————————————————
113MiB/372MiB ———————————————————
118MiB/372MiB ————————————————————
123MiB/372MiB —————————————————————
128MiB/372MiB ——————————————————————
133MiB/372MiB ———————————————————————
138MiB/372MiB ————————————————————————
143MiB/372MiB —————————————————————————
145MiB/372MiB —————————————————————————
150MiB/372MiB ——————————————————————————
155MiB/372MiB ———————————————————————————
159MiB/372MiB ———————————————————————————
164MiB/372MiB —————————————————————————————
168MiB/372MiB —————————————————————————————
174MiB/372MiB ——————————————————————————————
179MiB/372MiB ———————————————————————————————
182MiB/372MiB ————————————————————————————————
186MiB/372MiB —————————————————————————————————
190MiB/372MiB —————————————————————————————————
192MiB/372MiB —————————————————————————————————
196MiB/372MiB ——————————————————————————————————
201MiB/372MiB ———————————————————————————————————
206MiB/372MiB ————————————————————————————————————
210MiB/372MiB ————————————————————————————————————
216MiB/372MiB ——————————————————————————————————————
220MiB/372MiB ——————————————————————————————————————
225MiB/372MiB ———————————————————————————————————————
230MiB/372MiB ————————————————————————————————————————
234MiB/372MiB —————————————————————————————————————————
239MiB/372MiB ——————————————————————————————————————————
241MiB/372MiB ——————————————————————————————————————————
245MiB/372MiB ——————————————————————————————————————————
249MiB/372MiB ————————————————————————————————————————————
252MiB/372MiB ————————————————————————————————————————————
257MiB/372MiB —————————————————————————————————————————————
261MiB/372MiB ——————————————————————————————————————————————
265MiB/372MiB ——————————————————————————————————————————————
270MiB/372MiB ———————————————————————————————————————————————
275MiB/372MiB ————————————————————————————————————————————————
280MiB/372MiB —————————————————————————————————————————————————
285MiB/372MiB ——————————————————————————————————————————————————
286MiB/372MiB ——————————————————————————————————————————————————
290MiB/372MiB ———————————————————————————————————————————————————
295MiB/372MiB ————————————————————————————————————————————————————
300MiB/372MiB ————————————————————————————————————————————————————
303MiB/372MiB —————————————————————————————————————————————————————
309MiB/372MiB ——————————————————————————————————————————————————————
314MiB/372MiB ———————————————————————————————————————————————————————
319MiB/372MiB ————————————————————————————————————————————————————————
323MiB/372MiB ————————————————————————————————————————————————————————
327MiB/372MiB —————————————————————————————————————————————————————————
331MiB/372MiB ——————————————————————————————————————————————————————————
332MiB/372MiB ——————————————————————————————————————————————————————————
336MiB/372MiB ———————————————————————————————————————————————————————————
340MiB/372MiB ————————————————————————————————————————————————————————————
345MiB/372MiB ————————————————————————————————————————————————————————————
350MiB/372MiB ——————————————————————————————————————————————————————————————
355MiB/372MiB ——————————————————————————————————————————————————————————————
360MiB/372MiB ———————————————————————————————————————————————————————————————
364MiB/372MiB ————————————————————————————————————————————————————————————————
369MiB/372MiB —————————————————————————————————————————————————————————————————
Done ———————————————————————————————————————————————————————————————————————————
tar: /opt: Cannot open: No such file or directory
tar: Error is not recoverable: exiting now
Failed to extract wine!

Running purge_packages...
Allowing packages required by the Wine (x64) app to be uninstalled
These packages were: cabextract, p7zip-full
Purging the pi-apps-112b87c1 package...
The following packages will be REMOVED:
  cabextract* libmspack0* pi-apps-112b87c1*
0 upgraded, 0 newly installed, 3 to remove and 0 not upgraded.
After this operation, 160 kB disk space will be freed.
(Reading database ... 
(Reading database ... 5%
(Reading database ... 10%
(Reading database ... 15%
(Reading database ... 20%
(Reading database ... 25%
(Reading database ... 30%
(Reading database ... 35%
(Reading database ... 40%
(Reading database ... 45%
(Reading database ... 50%
(Reading database ... 55%
(Reading database ... 60%
(Reading database ... 65%
(Reading database ... 70%
(Reading database ... 75%
(Reading database ... 80%
(Reading database ... 85%
(Reading database ... 90%
(Reading database ... 95%
(Reading database ... 100%
(Reading database ... 329055 files and directories currently installed.)

Removing pi-apps-112b87c1 (1.0) ...

Removing cabextract (1.9-3) ...

Removing libmspack0:arm64 (0.11-1) ...

Apt finished.
All packages have been purged successfully.

Failed to install Wine (x64)!
◢◣Need help? Copy the ENTIRE terminal output or take a screenshot.
Please ask on Github: https://github.com/Botspot/pi-apps/issues/new/choose
Or on Discord: https://discord.gg/RXSTvaUvuu
```

**Comment by Botspot** [comment]
Thanks for reporting. I just got finished looking at this same report that you sent in using the 'send report' button.
In this case, it looks like the cause of your issues is that a system folder is missing `/opt/`. Do you remember deleting that folder?

We have had users report a missing /opt folder before issue before, so to fix it, a one-time repair command was added to pi-apps that checks for a missing /opt/ folder and adds creates it if missing. In your case the folder must have been removed after that one-time repair check was run, or something else went wrong.

Long story short, try running this command to create the missing /opt folder:
```
sudo mkdir /opt
```
This should solve the issue. After running that command, please retry installing Wine.

**Comment by FrenZy931** [comment]
Hello, thank you for your help. So i dont remember deleting opt folder, but when i added it there, it worked and i sucesfully installed Wine64 on my raspberry pi. Thank you very much for your help.

**Comment by theofficialgman** [comment]
@FrenZy931 can you upload or paste the contents of the file `~/pi-apps/data/runonce_hashes`

With those files contents I can check to make sure that the runonce that corrects `/opt` actually did run at some point in the past.

**Comment by FrenZy931** [comment]
So, for some reason, i cant upload the file, but i will paste it: 

19637fcc657af32d1b834e920bc54ee72f2b3076
d0de63dc3a6f0861fd29a61b772a1ce36f349ba8
4fc1fdfcc760e68d067f1bb9af22822393bfef0a
e1a455d3be745b9cb033c0d053532142b08fbc24
50a13d1138fca812c87131f8cda23d31762b1f6b
df7ad95f0e073fd66c146647cba094e9c66b2dc0
dfb57ecc75c43daf7c4592a1115f59ad5f04856d
88fea75739686e3bad8c5aa64f48bc913f188176
e48bf347e7d32a0a3149c3214fe124ccf1b77543
084bb057834363ce504ae6736c8abefaf82fa702
604f470e06084647eff30fe8a490a35e2ac7a751
65fc6637314ef4b072cfd87fdc3a6b561a2e1f09
9abe286d42630496e396750937ae31b4ae0df344
32257dab359557af25997b1dab042419e4c53cfb
3174f1fa3a14435eebd575d3c7f705ca80561bbe
34c4034ed9d04ac3f720faf452bc6a45ddc4b354
534195dab8025e28a6944508f79800d53a855f13
56963e4d7e55f079dcd6848682014a3bb97fbdc2
48e310f63054e8a5cc68f7280b96d71d19f1bc22
e5b75857d869a68bb1f5a84fcf8e97835a85ae6a
38510f22af0071c519799f3f177406c78b2832d9
8118c1f1e686fa42555188da47a769ceb6475f70

I hope this is what you need, becouse i have no idea what this is

**Comment by Botspot** [comment]
The `19637fcc657af32d1b834e920bc54ee72f2b3076` hash is a record indicating that the one-time fix to create the /opt folder has already been completed.
This indicates that your /opt folder mysteriously disappeared after you installed Pi-Apps.

**Comment by theofficialgman** [comment]
Not necessarily. While the runonce did complete it does not exit on individual command failure.
It's possible that 
`sudo_popup mkdir /opt` failed due to the user closing out of password prompts.

In a new terminal can you please paste and run
```
sudo_popup() { #just like sudo on passwordless systems like PiOS, but displays a password dialog otherwise. Avoids displaying a password prompt to an invisible terminal.
  if sudo -n true; then
    # sudo is available (within sudo timer) or passwordless
    sudo "$@"
  else
    # sudo is not available (not within sudo timer)
    pkexec "$@"
  fi
}

sudo echo "test"
```

**Comment by theofficialgman** [comment]
I am unable to reproduce any issue on my end. sudo_popup works correctly with passwordless sudo and sudo where a password is required and a popup requests the users password.

It is possible that the user hit cancel on the password prompt. The way this runonce is structured, it will NOT exit and error code on individual command failures.

**Comment by FrenZy931** [comment]
Hello, sorry but i wasnt home for 4 days, and i didn't had access to internet. I will run the comnand right when i will get back to my raspberry pi.

**Comment by hyppyp** [comment]
While updating Wine today, the process stuck with the window 'The Wine configuration in E:\home\pi\.wine is being updated, please wait...'. The message 'err:system:update_display_cache Failed to read display config' is showing perpetually.
System: Pi OS 64 Bullseye on a Pi400
I have the same issue on Pi OS 64 Buster on a Pi4 8gb (I know, not supported anymore). After breaking the process the desktop turns black (toolbar still ok). Can't restore the desktop, not even after cold reboot. System hangs.

**Comment by theofficialgman** [comment]
> While updating Wine today, the process stuck with the window 'The Wine configuration in E:\home\pi.wine is being updated, please wait...'. The message 'err:system:update_display_cache Failed to read display config' is showing perpetually. System: Pi OS 64 Bullseye on a Pi400 I have the same issue on Pi OS 64 Buster on a Pi4 8gb (I know, not supported anymore). After breaking the process the desktop turns black (toolbar still ok). Can't restore the desktop, not even after cold reboot. System hangs.

this is unrelated to this issue.

Regardless it is a box64 bug that affects all versions https://github.com/ptitSeb/box64/issues/1031 . it might be fixed in box64 master branch as of a few hours ago but we are waiting on the build here https://github.com/Pi-Apps-Coders/box64-debs/actions/runs/6667145327 to finish

**Comment by theofficialgman** [comment]
> Hello, sorry but i wasnt home for 4 days, and i didn't had access to internet. I will run the comnand right when i will get back to my raspberry pi.

@FrenZy931 bump
